namui-prebuilt/src/list_view.rs

```diff
@@ -0,0 +1,113 @@
+use crate::scroll_view;
+use namui::prelude::*;
+
+/// ListView is a vertical list view with fixed height items.
+#[derive(Debug)]
+pub struct ListView {
+    scroll_view: scroll_view::ScrollView,
+}
+
+pub struct Props<TItem, TIterator, TItems, TItemRender>
+where
+    TIterator: ExactSizeIterator<Item = TItem>,
+    TItems: IntoIterator<Item = TItem, IntoIter = TIterator>,
+    TItemRender: Fn(TItem, usize) -> RenderingTree,
+{
+    pub x: f32,
+    pub y: f32,
+    pub height: f32,
+    pub scroll_bar_width: f32,
+    pub item_wh: Wh<f32>,
+    pub items: TItems,
+    pub item_render: TItemRender,
+}
+
+pub enum Event {}
+
+impl ListView {
+    pub fn new() -> Self {
+        Self {
+            scroll_view: scroll_view::ScrollView::new(),
+        }
+    }
+    pub fn update(&mut self, event: &dyn std::any::Any) {
+        self.scroll_view.update(event);
+    }
+    pub fn render<TItem, TIterator, TItems, TItemRender>(
+        &self,
+        props: Props<TItem, TIterator, TItems, TItemRender>,
```

Comment:
Note that this props is not reference type unlike other props.
I think we should not use reference type for props because props struct is collection of parameter, and the purpose is named argument which rust doesn't support yet.

So if you want to move value, not reference, you cannot with `&Props` because it's reference type. But you can move value without Props.

To keep it same mood, I suggest not to use `&Props`, but just `Props`.

It needs to be discussion(https://github.com/NamseEnt/namseent/issues/167) so I just remove reference only here.

Reply:
I think it's okay.
Why don't we apply it to scroll_view too?

Reply:
Let me think about this problem more, then I will replace all later if I'm sure about this.